File: src/test/java/cz/vhromada/catalog/mapper/EpisodeMapperTest.java
package cz.vhromada.catalog.mapper;

import static org.assertj.core.api.Assertions.assertThat;

import cz.vhromada.catalog.entity.Episode;
import cz.vhromada.catalog.utils.EpisodeUtils;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;

/**
 * A class represents test for mapper between {@link cz.vhromada.catalog.domain.Episode} and {@link Episode}.
 *
 * @author Vladimir Hromada
 */
class EpisodeMapperTest {

    /**
     * Instance of {@link EpisodeMapper}
     */
    private EpisodeMapper mapper;

    /**
     * Initializes mapper.
     */
    @BeforeEach
    void setUp() {
        mapper = Mappers.getMapper(EpisodeMapper.class);
    }

    /**
     * Test method for {@link EpisodeMapper#map(Episode)}.
     */
    @Test
    void map() {
        final Episode episode = EpisodeUtils.newEpisode(1);
        final cz.vhromada.catalog.domain.Episode episodeDomain = mapper.map(episode);

        EpisodeUtils.assertEpisodeDeepEquals(episode, episodeDomain);
    }

    /**
     * Test method for {@link EpisodeMapper#map(Episode)} with null episode.
     */
    @Test
    void map_NullEpisode() {
        assertThat(mapper.map(null)).isNull();
    }


    /**
     * Test method for {@link EpisodeMapper#mapBack(cz.vhromada.catalog.domain.Episode)}.
     */
    @Test
    void mapBack() {
        final cz.vhromada.catalog.domain.Episode episodeDomain = EpisodeUtils.newEpisodeDomain(1);
        final Episode episode = mapper.mapBack(episodeDomain);

        EpisodeUtils.assertEpisodeDeepEquals(episode, episodeDomain);
    }

    /**
     * Test method for {@link EpisodeMapper#mapBack(cz.vhromada.catalog.domain.Episode)} with null episode.
     */
    @Test
    void mapBack_NullEpisode() {
        assertThat(mapper.mapBack(null)).isNull();
    }

}

File: catalog-web/src/test/kotlin/com/github/vhromada/catalog/web/mapper/EpisodeMapperTest.kt
package com.github.vhromada.catalog.web.mapper

import com.github.vhromada.catalog.entity.Episode
import com.github.vhromada.catalog.web.CatalogMapperTestConfiguration
import com.github.vhromada.catalog.web.common.EpisodeUtils
import com.github.vhromada.catalog.web.fo.EpisodeFO
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ContextConfiguration
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * A class represents test for mapper between [Episode] and [EpisodeFO].
 *
 * @author Vladimir Hromada
 */
@ExtendWith(SpringExtension::class)
@ContextConfiguration(classes = [CatalogMapperTestConfiguration::class])
class EpisodeMapperTest {

    /**
     * Mapper for episodes
     */
    @Autowired
    private lateinit var mapper: EpisodeMapper

    /**
     * Test method for [EpisodeMapper.map].
     */
    @Test
    fun map() {
        val episode = EpisodeUtils.getEpisode()

        val episodeFO = mapper.map(episode)

        EpisodeUtils.assertEpisodeDeepEquals(episodeFO, episode)
    }

    /**
     * Test method for [EpisodeMapper.mapBack].
     */
    @Test
    fun mapBack() {
        val episodeFO = EpisodeUtils.getEpisodeFO()

        val episode = mapper.mapBack(episodeFO)

        EpisodeUtils.assertEpisodeDeepEquals(episodeFO, episode)
    }

}
